core/arch/arm/plat-imx/psci.c

```diff
@@ -67,3 +71,60 @@ int psci_cpu_on(uint32_t core_idx, uint32_t entry,
 
 	return PSCI_RET_SUCCESS;
 }
+
+int psci_cpu_off(void)
+{
+	uint32_t core_id;
+
+	core_id = get_core_pos();
+
+	DMSG("core_id: %d\n", core_id);
+
+	psci_armv7_cpu_off();
+
+	imx_set_src_gpr(core_id, UINT32_MAX);
+
+	thread_mask_exceptions(THREAD_EXCP_ALL);
+
+	while (true)
+		wfi();
+
+	return PSCI_RET_INTERNAL_FAILURE;
+}
+
+int psci_affinity_info(uint32_t affinity,
+		       uint32_t lowest_affnity_level __unused)
+{
+	vaddr_t va = core_mmu_get_va(SRC_BASE, MEM_AREA_IO_SEC);
+	vaddr_t gpr5 = core_mmu_get_va(IOMUXC_BASE, MEM_AREA_IO_SEC) +
+		IOMUXC_GPR5_OFFSET;
```

Comment:
linux checkpatch does not report warning on this.  To open (, need to align, but here ( is closed with ), no need to align?